### ISSUE TITLE
Support Windows ARM64

### ISSUE THREAD

**Issue by Rono0365** [oven-sh/bun#9824]
### What version of Bun is running?

bun 1.1

### What steps can reproduce the bug?

install Failed:
Bun for Windows is currently only available for x86 64-bit Windows.

1

**Comment by aishuaikang** [comment]
I am also waiting for a solution

**Comment by Baiyuetribe** [comment]
+1

**Comment by benpsnyder** [comment]
+1 👍  ... https://x.com/benpsnyder/status/1777155495003934877

**Comment by sky0matic** [comment]
With the recent push of Microsoft/Snapdragon with X Plus/Elite processors, this is something that should be supported.

Getting my laptop with one this week, I wouldn't mind looking at what it would take, however I'm new to building stuff from sources and wouldn't know where to start 😅

**Comment by sky0matic** [comment]
So got my laptop today.
Like mentionned, it doesn't install in Windows because it's not supported.

Though you can still use `bun` inside the WSL which seems to work fine there.

**Comment by Izzur** [comment]
I tried installing bun v1.1.27 with PowerShell 7.4.5 on Windows 11/24H2 (Qualcomm Snapdragon X Plus). So far it works normal. Also tried `bun create elysia` and run it, also works.

**Comment by itpropro** [comment]
Any updates on this? Is there a native Windows ARM64 binary on the horizon?

**Comment by donjjones** [comment]
> If you use the Windows x64 baseline build with an x64 emulator it should work fine for now

Can you elaborate on this?  Would love to be able to run Bun on my Snapdragon X ARM system

**Comment by Snazzie** [comment]
> > If you use the Windows x64 baseline build with an x64 emulator it should work fine for now
> 
> Can anyone elaborate on this? Would love to be able to run Bun on my Snapdragon X ARM system

it means it will run on snapdragon machine. just not very fast.

**Comment by sanathusk** [comment]
+1

**Comment by agarny** [comment]
I thought I would try the x64 version of `bun` on my Windows ARM test machine, but... it crashes. So, it looks like we just can't use `bun` on Windows ARM at this stage? Going to have to go back to `pnpm`. :(

**Comment by agarny** [comment]
Ok, someone on the `bun` Discord (thanks @RiskyMH!) got me to check things further. Bottom line: to use `bun` on Windows ARM, one needs to use the baseline x64 version of `bun` and not just its x64 version. The difference is that the baseline version is for machines that don't support [AVX2](https://en.wikipedia.org/wiki/Advanced_Vector_Extensions) instructions (see [here](https://github.com/oven-sh/bun/tree/main/packages/bun-release/npm/%40oven/bun-windows-x64-baseline)).

So, although there is no Windows ARM version of `bun`, we can use its baseline x64 version. This means that for GitHub Actions, we can install `bun` on Windows ARM by doing something like:
```
curl -L https://github.com/oven-sh/bun/releases/latest/download/bun-windows-x64-baseline.zip -o bun.zip
unzip -j bun.zip -d $env:USERPROFILE\.bun
del bun.zip
echo "$env:USERPROFILE\.bun" >> $env:GITHUB_PATH
```

From there, your can use `bun` on Windows ARM, but... in my case, I need to "manually" install some native packages using (`npm install @rollup/rollup-win32-arm64-msvc @tailwindcss/oxide-win32-arm64-msvc lightningcss-win32-arm64-msvc` in my case).

So, yeah, far from ideal, but at least it works... albeit my CI/CD is now 50% slower on Windows ARM (compared with `pnpm`). 😕

**Comment by Psychlist1972** [comment]
Hi All

It's unclear to me what's left to do here, and if there are any blockers for Arm64 builds for Windows. Is there a summary of open issues for Arm64 anywhere? 

Quite a few companies and projects depend on Bun, so I'm looking to get a handle on the size of the remaining work.

Any ETA for Arm64 Windows releases if not blocked?

Pete
Microsoft

**Comment by talynone** [comment]
Looks like we're waiting on this: https://github.com/oven-sh/bun/pull/26746

**Comment by ProgrammerIn-wonderland** [comment]
hopefully this is over soon <img width="1958" height="526" alt="Image" src="https://github.com/user-attachments/assets/df8e6dce-6be5-4246-828c-1241379124fc" />

**Comment by talynone** [comment]
> I am using `oven-sh/setup-bun@v2` to install Bun on Windows ARM in my GitHub workflow and I am getting the following:
>  
> Am I missing something?

Yes, that step is trying to install bun-windows-x64-baseline.zip, that's the x64 version, you want the ARM64 version bun-windows-aarch64.zip. That GitHub action probably hasn't been updated with information for the ARM version. 

You can override the download url manually:
```

- uses: oven-sh/setup-bun@v2
  with:
    bun-version: latest
    # Manually specify the URL for the architecture you want (e.g., linux-x64)
    bun-download-url: "https://github.com/oven-sh/bun/releases/latest/download/bun-windows-aarch64.zip"
```

**Comment by duncanmak** [comment]
The binaries are up for Windows ARM64.

`setup-bun@v2` now supports `windows-11-arm`: see https://github.com/oven-sh/setup-bun/issues/168

winget now has both x64 and arm64 binaries: https://github.com/microsoft/winget-pkgs/pull/343115


I think this issue can now be closed.

**Comment by agarny** [comment]
Indeed, all working fine for me! Thanks to those who made this possible, much appreciated!
